packages/bits/src/lib/repeat/repeat.component.ts

```diff
@@ -349,49 +400,81 @@ export class RepeatComponent<T extends IRepeatItem = any> implements OnInit, OnD
             return;
-        } else if (this.selectionMode === RepeatSelectionMode.multi) {
-            event.preventDefault(); // TODO: add reasoning for this
+        }
+
+        // prevent handling both from row and input
+        event.stopImmediatePropagation();
+        event.stopPropagation();
+        event.preventDefault();
+
+        if (this.selectionMode === RepeatSelectionMode.multi) {
             this.selectionHasChanged = true;
             this.multiSelectionChanged(item);
-        } else {
-            if (this.selectionMode === RepeatSelectionMode.singleWithRequiredSelection ||
-                this.selectionMode === RepeatSelectionMode.radio) {
+            return;
+        }
+
+        if (
+            this.selectionMode ===
+                RepeatSelectionMode.singleWithRequiredSelection ||
+            this.selectionMode === RepeatSelectionMode.radio
+        ) {
+            this.selection = [item];
+        }
+        if (
+            this.selectionMode === RepeatSelectionMode.single ||
+            this.selectionMode ===
+                RepeatSelectionMode.radioWithNonRequiredSelection
+        ) {
+            if (this.selection.indexOf(item) !== -1) {
+                this.selection = [];
+            } else {
                 this.selection = [item];
             }
-            if (this.selectionMode === RepeatSelectionMode.single ||
-                this.selectionMode === RepeatSelectionMode.radioWithNonRequiredSelection) {
-                if (this.selection.indexOf(item) !== -1) {
-                    this.selection = [];
-                } else {
-                    this.selection = [item];
-                }
-            }
-            this.changeDetector.markForCheck();
-            this.selectionHasChanged = true;
-            this.selectionChange.emit(this.selection);
-            return;
         }
+        this.changeDetector.markForCheck();
+        this.selectionHasChanged = true;
+        this.selectionChange.emit(this.selection);
     }
 
+
     /**
      * nui-repeat-item selection change handler
      * @param item selected nui-repeat-item
      */
-    public multiSelectionChanged(item: T) {
+    public multiSelectionChanged(item: T): void {
         if (this.selection.indexOf(item) === -1) {
             this.selection = [...this.selection, item];
         } else {
-            this.selection = this.selection.filter(selectionItem => selectionItem !== item);
+            this.selection = this.selection.filter(
+                (selectionItem) => selectionItem !== item
+            );
         }
         this.selectionChange.emit(this.selection);
     }
 
+    public isMultiselectable(): boolean {
+        return this.selectionMode === RepeatSelectionMode.multi;
+    }
+
     /* START - ITEM BEHAVIOUR DECIDERS */
+    public isItemClickable(item: T): boolean {
+        return !this.preventRowClick && !this.isItemDisabled(item);
+    }
+
+    public isItemSelectable(item: T): boolean {
```

Comment:
do we need this BEHAVIOUR DECIDER? 
if yes then we can use it in ln 467 too :P

Reply:
I didn't want to tie it together too much, yet have some distinct name, but I can remove it